File: video-service/src/main/java/tv/comnata/videoservice/services/VideoResolution.java
package tv.comnata.videoservice.services;

import org.jetbrains.annotations.NotNull;

public class VideoResolution implements Comparable<VideoResolution> {
    private final int width;
    private final int height;

    public VideoResolution(String resolution) {
        String[] r = resolution.split("x");
        this.width = Integer.parseInt(r[0]);
        this.height = Integer.parseInt(r[1]);
    }

    public VideoResolution(int width, int height) {
        this.width = width;
        this.height = height;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    @Override
    public String toString() {
        return width + "x" + height;
    }

    @Override
    public int compareTo(@NotNull VideoResolution o) {
        return this.width - o.width;
    }
}

File: video-service/src/main/java/tv/comnata/videoservice/services/VideoResolution.kt
package tv.comnata.videoservice.services

class VideoResolution : Comparable<VideoResolution> {
    val width: Int
    val height: Int

    constructor(resolution: String) {
        val r = resolution.split("x".toRegex()).toTypedArray()
        width = r[0].toInt()
        height = r[1].toInt()
    }

    constructor(width: Int, height: Int) {
        this.width = width
        this.height = height
    }

    private fun getBandwidth(): Int {
        return when (height) {
            240 -> 246440
            360 -> 460560
            480 -> 836280
            720 -> 2149280
            1080 -> 6221600
            else -> 6221600
        }
    }

    fun getBaseFileText(): String {
        return "#EXT-X-STREAM-INF:" +
                "PROGRAM-ID=1," +
                "BANDWIDTH=${getBandwidth()}," +
                "RESOLUTION=${width}x$height," +
                "NAME=\"$height\"\n" +
                "${height}p/video.m3u8"
    }

    override fun toString(): String {
        return width.toString() + "x" + height
    }

    override fun compareTo(other: VideoResolution): Int {
        return width - other.width
    }
}